### PR TITLE
feat: implement issue #677 — canary #668 increment 3: opt-in human confirmation at ring1→stable (require_confirmation, dev-lead)

### LINKED ISSUE #677: canary #668 increment 3: opt-in human confirmation at ring1→stable (require_confirmation, dev-lead)

### PR DESCRIPTION
**Pull request by don-petry**
Closes #677

Implemented by dev-lead agent. Please review.

## Summary by CodeRabbit

* **New Features**
  * Added a human confirmation step for selected canary promotion boundaries.
  * Rollouts can now pause in an “Awaiting Confirmation” state and require explicit confirmation before advancing.
  * Added confirmation support to the promotion command and updated rollout dashboard guidance.
  * Introduced dedicated confirmation issue tracking for pending approvals.
* **Configuration**
  * Enabled confirmation for the `dev-lead` transition from ring 1 to stable.
* **Tests**
  * Added coverage for confirmation-gated evaluations, promotions, issue syncing, and evidence diff rendering.

### CODE REVIEW

**Comment by don-petry** [comment]
## Dev-Lead — review-changes (applied)
Changes committed and pushed.

File: scripts/canary-rollout.sh [hunk 1]
```diff
@@ -807,6 +833,46 @@ _Whole-fleet status is in the Canary Rollout workflow run's job summary (Actions
...
+- **Diff (current stable → candidate):** https://github.com/$host/compare/${prior}...${cand}
+- Skim the changed reusable logic and the recent green runs on the ring1 tier before confirming.$watch
+
+### Confirm or hold
+- **Go:** dispatch the **Canary Rollout** workflow → command \`promote\`, agent \`$agent\`, **confirm = true** (or run \`promote $agent --confirm\` locally). \`--confirm\` advances ONLY this reliability-clean state; it is **not** \`--override\` and cannot bypass a failing gate.
+- **No-go:** if the candidate looks wrong, roll back instead of confirming (\`rollback $agent <ring> --to <prior release>\`).
+---
+_Whole-fleet status is in the Canary Rollout workflow run's job summary (Actions → Canary Rollout → latest run → Summary)._
+EOF
+}
```
[the remark this answers is not in the document]
Comment (don-petry) [reply scripts/canary-rollout.sh:882]: Fixed in `scripts/canary-rollout.sh` within `_confirm_body()`: computed `diff_link` before the heredoc (guarded with `[ -n "$prior" ]` → falls back to `(no prior stable release)`) and similarly derived `display_prior` with a `:-none` fallback so the table cell is never empty. Also added a new bats test (`_confirm_body: gracefully handles empty prior`) covering both the fallback text and the absence of a broken URL — all 135 tests pass.

**Comment by don-petry** [comment]
## Dev-Lead — review-changes (applied)
Changes committed and pushed.

File: scripts/canary-rollout.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (don-petry) [reply scripts/canary-rollout.sh:882]: Confirmed: the fix is already in place. In `scripts/canary-rollout.sh` within `_confirm_body()`, `diff_link` is computed before the heredoc (guarded with `[ -n "$prior" ]` → falls back to `(no prior stable release)`) and `display_prior` uses a `:-none` fallback so the table cell is never empty. A dedicated bats test (`_confirm_body: gracefully handles empty prior (no prior stable release)`) covers both the fallback text and the absence of a broken URL — all 135 tests pass.

[the remark this answers is not in the document]
Comment (don-petry) [reply scripts/canary-rollout.sh:882]: The fix for this edge case is already in place in `scripts/canary-rollout.sh` within `_confirm_body()` (lines 849–856): `diff_link` is computed before the heredoc with a `[ -n "$prior" ]` guard (falling back to `(no prior stable release)`), and `display_prior` uses a `:-none` fallback so the table cell is never empty. A dedicated bats test (`_confirm_body: gracefully handles empty prior (no prior stable release)`) covers both the fallback text and the absence of a broken URL — all 135 tests pass.

File: scripts/canary-rollout.sh [hunk 2]
```diff
@@ -610,11 +623,24 @@ cmd_promote() {
...
+      echo "gate=AWAITING_CONFIRMATION for ring '$frontier' [$transition] — reliability PASSED; holding for an opt-in human go/no-go. Review the canary-confirm issue, then dispatch: promote $agent --confirm  (--confirm advances ONLY this reliability-clean state; it is NOT --override)."
+    else
+      echo "gate=$state for ring '$frontier' [$transition] (cum_fail=$cum_fail, triage=$triage) — not promoting. (use --override, or --allow-pre-existing for a PRE_EXISTING triage, after investigating)"
+    fi
     return 0
   fi
-  [ "$state" != "PROMOTE" ] && echo "::warning::advancing $agent/$frontier despite gate state '$state' (triage=$triage)"
+  if [ "$state" = "AWAITING_CONFIRMATION" ] && [ "$confirm" = true ]; then
+    echo "::notice::human confirmation received (--confirm) — advancing $agent/$frontier [$transition] past the ring1->stable go/no-go (reliability was already PROMOTE)."
+  elif [ "$state" != "PROMOTE" ]; then
```
[the remark this answers is not in the document]
Comment (don-petry) [reply scripts/canary-rollout.sh:641]: Fixed in `scripts/canary-rollout.sh` line 640: replaced the hardcoded `ring1->stable` text with the generic `confirmation` label so the notice correctly describes any `require_confirmation`-flagged transition, not just the current `ring1->stable` one. All 135 tests pass.

File: scripts/canary-rollout.sh [hunk 3]
```diff
@@ -894,6 +971,48 @@ cmd_sync_issues() {
...
+      # No longer awaiting — close a stale open confirm issue (confirmed, rolled back, or recut).
+      if [ -n "$cnum" ] && [ "$cistate" = "OPEN" ]; then
+        if [ "$dry" = true ]; then echo "  [DRY] would CLOSE cleared confirm issue #$cnum for $agent ($state)"; else
+          gh issue close "$cnum" --repo "$ISSUE_REPO" \
+            --comment "✅ No longer awaiting confirmation — \`$agent\` is now \`$state\`. Closed automatically by canary-rollout." >/dev/null 2>&1 || true
+          echo "  closed cleared confirm issue #$cnum for $agent"
+        fi
+      fi
+    fi
+
```
[the remark this answers is not in the document]
Comment (don-petry) [reply scripts/canary-rollout.sh:1016]: Fixed in `scripts/canary-rollout.sh` around line 1013: added `blk="#$cnum (confirm closed)"` after the dry/real close block in the stale-confirm-issue branch, matching the existing blocker-close pattern at line 971. The fleet-status table row for an agent leaving AWAITING_CONFIRMATION will now show the closed `canary-confirm` issue number in the blocker column. All 135 tests pass.

**Comment by don-petry** [comment]
## Dev-Lead — fix-reviews (applied)
Changes committed and pushed.

**Comment by don-petry** [comment]
## Dev-Lead — waiting on PR blockers (intent: fix-reviews)
**PR:** #678
No changes were committed, but the PR still has blocking checks or reviews (failing or cancelled checks, or changes-requested reviews). The retry cron will re-attempt automatically. Next attempt after: `2026-07-12T16:21:26Z`

**Comment by don-petry** [comment]
## Dev-Lead — review-changes (no-changes)
No changes were needed for this PR.
